plugins/cpp_reparse/service/src/typespecialmembers.cpp

```diff
@@ -0,0 +1,1502 @@
+
+      if (diagLevel_ == DiagnosticsEngine::Level::Error)
+      {
+        // Try to match against the special member marker in the handcrafted
+        // usage lines that could have emitted the error.
+        auto smMarker = line.find("/* SM: ");
+        if (smMarker == std::string::npos)
+          // An error was emitted from the original file, not the crafted code.
+          return;
+
+        smMarker += 7; // 7 == std::string("/* SM: ").size()
+        auto smMarkerEnd = line.find(" */", smMarker + 1);
+        std::string tail = line.substr(smMarker, smMarkerEnd - smMarker);
+
+        _shouldCaptureDiagnostic = true;
+        _currentCapturingAnnotation = tail;
+
+        // However, the error itself should not be captured as it is in temp
+        // code.
+        return;
+      }
+
+      assert(diagLevel_ == DiagnosticsEngine::Level::Note &&
+             "Algorithm failure: at this point, the diagnostic should've been "
+             "a note.");
+
+      SmallString<128> message;
+      info_.FormatDiagnostic(message);
+
+      PresumedLoc presumed = manager.getPresumedLoc(info_.getLocation());
+      std::string presumedFilename = presumed.isValid() ?
+                                     presumed.getFilename() : "<unknown>";
+      std::size_t presumedLineNo = presumed.isValid() ?
+                                   presumed.getLine() : 0;
+
+      _diagnostics[_currentCapturingAnnotation].emplace_back(
+        std::move(presumedFilename), presumedLineNo,
+        std::move(line), message.c_str());
+    }
+
+    const std::vector<DiagnosticMessage>& getDiagnostics(
+      const std::string& group_)
+    {
+      return _diagnostics[group_];
+    }
+
+  private:
+
+    bool _shouldCaptureDiagnostic = false;
+
+    /**
+     * The special member annotation string (e.g. "CopyCtor") that is set to be
+     * captured right now.
+     * This is only applicable if _shouldCaptureDiagnostic is true.
+     */
+    std::string _currentCapturingAnnotation;
+
+    /**
+     * Maps captured diagnostic to the special member annotations found in the
+     * crafted code.
+     */
+    std::map<std::string, std::vector<DiagnosticMessage>> _diagnostics;
+  };
+
+public:
+
+  /**
+   * Represents a single special member of a record in visitation.
+   */
+  struct SpecialMember
+  {
+    /**
+     * A special member's full source text can be composed of multiple strings,
+     * instances of TextComponent represent one such substring.
+     */
+    struct TextComponent
+    {
+      enum struct Mapping
+      {
+        Unmapped, /*!< The source code of the special member is generated by
+          the visitor, it is not found explicitly in any file. */
+        OnFile, /*!< The source code is explicitly available in the file and
+          range specified by the instance. */
+        ThroughDeclaration /*!< The location of the source code is unknown, but
+          can be resolved via CodeCompass' database. */
```

Comment:
You are misunderstanding the whole point of the pull request it seems. Existence of AST nodes for generated functions are of no concern of this feature (although currently they exist). This pull request generates a non-existent and nowhere stored *body* for the functions. `Unmapped` means that the body of the function is not in any file (because it is a compiler-generated function), and is to be, or was, generated by the visitor implemented in this PR.